packages/twenty-server/src/engine/workspace-manager/dev-seeder/data/services/timeline-activity-seeder.service.ts

```diff
@@ -0,0 +1,395 @@
+          const linkedObjectMetadataId =
+            type === 'note' ? noteMetadataId : taskMetadataId;
+          const linkedActivities = this.createLinkedTimelineActivities(
+            type,
+            seed,
+            index,
+            activityIndex,
+            linkedObjectMetadataId,
+          );
+
+          timelineActivities.push(...linkedActivities);
+          activityIndex += linkedActivities.length;
+        }
+      });
+    });
+
+    if (timelineActivities.length > 0) {
+      const batchSize = 100;
+      const timelineActivityBatches = chunk(timelineActivities, batchSize);
+
+      for (const batch of timelineActivityBatches) {
+        await entityManager
+          .createQueryBuilder(undefined, undefined, undefined, {
+            shouldBypassPermissionChecks: true,
+          })
+          .insert()
+          .into(`${schemaName}.timelineActivity`, [
+            'id',
+            'name',
+            'properties',
+            'linkedRecordCachedName',
+            'linkedRecordId',
+            'linkedObjectMetadataId',
+            'workspaceMemberId',
+            'companyId',
+            'personId',
+            'noteId',
+            'taskId',
+            'opportunityId',
+            'createdAt',
+            'updatedAt',
+            'happensAt',
+          ])
+          .orIgnore()
+          .values(batch)
+          .execute();
+      }
+    }
+  }
+
+  private createTimelineActivity(
+    entityType: string,
+    entitySeed: Record<string, unknown>,
+    index: number,
+  ): Record<string, unknown> {
+    const generateTimelineActivityId = (type: string, idx: number): string => {
+      const prefix = '20202020';
+      const entityCodes: Record<string, string> = {
+        company: '0001',
+        person: '0201',
+        note: '0601',
+        task: '0651',
+        opportunity: '0851',
+      };
+      const code = entityCodes[type] || '0000';
+      const paddedIndex = String(idx).padStart(4, '0');
+
+      return `${prefix}-${code}-4000-8001-${paddedIndex}00000001`;
+    };
+
+    const creationDate = new Date().toISOString();
+
+    const timelineActivity: Record<string, unknown> = {
+      id: generateTimelineActivityId(entityType, index + 1),
+      name: `${entityType}.created`,
+      properties: JSON.stringify({
+        after: this.getEntityProperties(entityType, entitySeed),
+      }),
+      linkedRecordCachedName: '',
+      linkedRecordId: entitySeed.id,
+      linkedObjectMetadataId: null,
+      workspaceMemberId: WORKSPACE_MEMBER_DATA_SEED_IDS.TIM,
+      companyId: null,
+      personId: null,
+      noteId: null,
+      taskId: null,
+      opportunityId: null,
+      createdAt: creationDate,
+      updatedAt: creationDate,
+      happensAt: creationDate,
+    };
+
+    timelineActivity[`${entityType}Id`] = entitySeed.id;
+
+    return timelineActivity;
+  }
+
+  private getEntityProperties(
+    type: string,
+    seed: Record<string, unknown>,
```

Comment:
when we have multiple parameters to a function, let's use an object and deconstruct, this eases the devXP